types/any_type.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+// AnyType to represent AnyType, this type is to indicate "any" type of argument
+// This type is not used in function invocation. It is only used in function definition
+type AnyType struct {
+	Name        string
+	Nullability Nullability
+}
+
+func (*AnyType) isRootRef() {}
+func (m *AnyType) WithNullability(nullability Nullability) Type {
+	m.Nullability = nullability
+	return m
```

Comment:
Shouldn't this return a copy with the changed nullability rather than change the one being used? It feels like this should be a value receiver, not a pointer receiver

Reply:
fixed it